include/odp/api/spec/threshold.h

```diff
@@ -0,0 +1,89 @@
+/* Copyright (c) 2017, Linaro Limited
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier:     BSD-3-Clause
+ */
+
+/**
+ * @file
+ *
+ * ODP threshold descriptor
+ */
+
+#ifndef ODP_API_THRESHOLD_H_
+#define ODP_API_THRESHOLD_H_
+#include <odp/visibility_begin.h>
+#include <odp/api/std_types.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * Supported threshold types
+ *
+ * Supported threshold types in a bit field structure.
+ */
+typedef union odp_threshold_type_t {
+	struct {
+		/** Percentage of the total size of pool or queue */
+		uint8_t percent:1;
+		/* Number of transient packets */
+		uint8_t pkt_cnt:1;
+		/* Total size of all transient packets in bytes */
+		uint8_t pkt_size:1;
+	};
+	/** All bits of the bit field structure */
+	uint8_t all_bits;
+} odp_threshold_type_t;
+
+/**
+ * ODP Threshold types
+ *
+ * Different types of threshold measurements
+ */
+typedef	enum {
+	/** Percentage of the total size of pool or queue */
+	odp_percent_e,
+	/** Number of transient packets */
+	odp_pkt_count_e,
+	/** Total size of all transient packets in bytes */
+	odp_pkt_size_e
+} odp_threshold_type_e;
```

Comment:
I felt using a bitfield is more useful in exposing the platform capability whereas using an enum makes more sense during configuration. I am open for suggestions along these lines.

Reply:
the enum: odp_threshold_type_t
the bitfield: odp_threshold_types_t

The enum values need to be UPPERCASE and contain a common prefix: ODP_THLD_PERCENT, ODP_THLD_PACKET, ODP_THLD_BYTE

If bitfield is only needed in one place (one capability struct) it could be defined there only (no typedef needed).

Reply:
Another option for enums: ODP_THRESHOLD_PCT, ODP_THRESHOLD_PKT, ODP_THRESHOLD_BYTE